administration/src/testing/render.tsx

```diff
@@ -43,3 +30,78 @@ export const renderWithTranslation = (
 
   return rawRender(ui, { wrapper })
 }
+
+type CustomRenderOptions = {
+  projectConfig?: ProjectConfig
+  theme?: boolean
+  translation?: boolean
+  router?: boolean
+  localization?: boolean
+  snackbar?: boolean
+}
+
+export const renderWithOptions = (
+  ui: ReactElement,
+  options: RenderOptions & CustomRenderOptions = {}
+): RenderResult => {
+  const {
+    projectConfig,
+    theme = false,
+    translation = false,
+    router = false,
+    localization = false,
+    snackbar = false,
+    wrapper,
+  } = options
+  const wrappers: ((props: { children: ReactNode }) => ReactNode)[] = []
+  if (wrapper) {
+    const CustomWrapper = wrapper
+    const customWrapper = (props: { children: ReactNode }) => <CustomWrapper {...props} />
+    wrappers.push(customWrapper)
+  }
```

Comment:
🔧  I think this is a case where React's `createElement()` should be used:

```suggestion
  const wrappers: ((props: { children: ReactNode }) => ReactNode)[] = []

  if (wrapper) {
    wrappers.push((props: { children: ReactNode }) => createElement(wrapper, props, null))
  }
```